python/ray/util/ml_utils/checkpoint_manager.py

```diff
@@ -0,0 +1,445 @@
+import copy
+import enum
+import gc
+import heapq
+import logging
+import numbers
+import os
+import shutil
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Optional, Dict, Union, Callable, Tuple, List, Any
+
+import ray
+from ray.ml import Checkpoint
+from ray.tune.result import NODE_IP
+from ray.util import PublicAPI
+from ray.util.annotations import DeveloperAPI
+from ray.util.ml_utils.util import is_nan
+
+MAX = "max"
+MIN = "min"
+
+logger = logging.getLogger(__name__)
+
+
+class CheckpointStorage(enum.Enum):
+    MEMORY = enum.auto()
+    PERSISTENT = enum.auto()
+
+
+@DeveloperAPI
+class TrackedCheckpoint:
```

Comment:
Can we keep these classes private and inside `ray/tune`? I don't see a reason why as a user or developer I should ever need to know about these.